shongo-controller/src/main/java/cz/cesnet/shongo/controller/booking/resource/Tag.java

```diff
@@ -2,27 +2,63 @@
 
 import cz.cesnet.shongo.SimplePersistentObject;
 import cz.cesnet.shongo.api.AbstractComplexType;
+import cz.cesnet.shongo.controller.api.TagType;
 import cz.cesnet.shongo.controller.booking.ObjectIdentifier;
+import org.hibernate.annotations.Type;
 
 import javax.persistence.Column;
 import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
 
 /**
  * @author: Ondřej Pavelka <pavelka@cesnet.cz>
  */
 @Entity
 public class Tag extends SimplePersistentObject {
+
+    @Column(unique = true)
     private String name;
 
-    @Column(length = AbstractComplexType.DEFAULT_COLUMN_LENGTH, unique = true)
-    public String getName() {
+    private TagType type;
+
+    private String data;
+
+    public String getName()
+    {
         return name;
     }
 
-    public void setName(String name) {
+    public void setName(String name)
+    {
         this.name = name;
     }
 
+    @Type(type = "pgsql_enum")
```

Comment:
Bude stačiť @Enumerated(EnumType.STRING) a @Column od JPA na definovanie typu. 
Ten @Type(type = "pgsql_enum") je duplicita.

Reply:
Bolo to potrebne ked som vytvaral enum aj v postgrese. Ale pre konzistenciu som to zmenil a odstranil.